src/menelaus/ensemble.py

```diff
@@ -98,8 +188,16 @@ def select_data(self, data, det_key: str):
         #           are mixed in
         ret = data.copy()
         if self.columns:
-            cols = self.columns[det_key]
-            ret = data[cols]
+            if det_key in self.columns:
```

Comment:
I think either this or `Ensemble.update` need a fallback where columns are unspecified, because `X` is not used -- and `Ensemble.update` probably needs to not require `X`.

Supposing the ensemble only contains detectors that require `y_true, y_pred`, the ensemble still demands `X` is specified along with a key in the `columns` dictionary.

The latter could be something we do by filling in on `init` for convenience wherever there's no key, like
```
self.columns = dict.fromkeys(self.detectors, None)
self.columns = {**self.columns, columns}
```

Reply:
On second thought, I like the `defaultdict` idea, and handling the issue you raise in general @tms-bananaquit, but I've made a note (converting to issue when I can) about refactoring this piece entirely.

We probably want to move towards column selector functions being passed in, to reduce the type checking strain on the menelaus side. Which makes me want to ignore this piece for the time being.

```python
selector1 = lambda df: df[['temperature', 'dew_point']]     # or whatever the pandas syntax is
selector2 = lambda ar: ar[:1,:]                             # or whatever the numpy syntax is

# ensemble doesn't need to do any type-based filtering! it's the user's fault hehe
ensemble = Ensemble(detectors, evaluator, selectors)   
```

Reply:
Yeah, I like this solution, if we couple it with handling for `selectors=None`.

For the simplest case, where we just want a univariate detector on `temperature`, if we set this up so that 

```
ensemble = Ensemble(detectors, evaluator, None)
ensemble.update(X=X.loc['temperature']
```
is valid, then that seems like a nice balance to me. If they want to run a data detector on the whole `X`, they can just pass all of it with no selector. If they're trying to do something more elaborate where they ensemble univariate detectors that each need to watch a separate column, then I don't think it's unreasonable to think they can manage lambdas.

Reply:
Yes, the univariate case is one of the reasons I brought up selectors -- it introduces a whole type checking thing that I would want to avoid. 

I'll also mention #99 here because maybe we actually _**do**_ want to avoid putting the ensemble code as the only README example, since it has all these features we want to draw attention to in examples, and also because the README is currently split nicely with a concept and data drift example.

Reply:
Oh, my reading of that issue was that we'd add a third simple example demonstrating that ensembles can be set up, rather than replacing the existing examples with ensembles themselves. Including something like that and then directing to the notebooks as more detailed examples seems reasonable.